mymodule/sky_sim.py

```diff
@@ -27,18 +26,27 @@ def make_positions():
     for i in range(NSRC):
         ras.append(ra + uniform(-1,1))
         decs.append(dec + uniform(-1,1))
+    
+    # apply our filter
+    ras, decs = crop_to_circle(ras, decs, ra, dec, 1)
     return ras, decs
 
 
-def clip_to_radius():
-    return
+def crop_to_circle(ras, decs, ref_ra, ref_dec, radius):
+    ra_out = []
+    dec_out = []
+    for i in range(len(ras)):
+        if (ras[i]-ref_ra)**2 + (decs[i]-ref_dec)**2 <= radius**2:
```

Comment:
This filter means we get less than NSRC sources in our catalogue. Could you consider an alternative approach?

Reply:
Yeah no worries